lib/signature-ot1.js

```diff
@@ -0,0 +1,247 @@
+ *
+ * Use this as middleware in your route files.
+ */
+
+/**
+ * @typedef {Object} validateSignatureMiddleware~parsedHeader
+ * @prop {string} method "OT1-HMAC-SHA256-HEX"
+ * @prop {string} accessCode
+ * @prop {Array.<string>} signedHeaders
+ * @prop {string} signature
+ */
+
+/**
+ * @typedef {Object} validateSignatureMiddleware~signatureDescription
+ * @property {string} algorithm Key generation algorithm
+ * @property {string} encoding How the resulting signature is encoded
+ * @property {string} method How the key is generated (HMAC)
+ * @property {string} signature The trimmed and standardized signature
+ * @property {string} type Signature type (OT1)
+ */
+
+module.exports = (accessCodeManager, errorResponse, hash, promise, util) => {
+    /**
+     * Builds the large string to sign.
+     *
+     * @param {Restify~Request} req
+     * @param {Array.<string>} signedHeaders
+     * @return {string}
+     */
+    function buildSigningContent(req, signedHeaders) {
+        var headerName, i, result, value;
+
+        result = [
+            req.method,
+            req.getPath(),
+            req.getQuery()
+        ];
+
+        for (i = 0; i < signedHeaders.length; i += 1) {
+            headerName = signedHeaders[i];
+            value = req.headers[headerName].trim();
+
+            if (headerName === "host") {
+                // The Host header must be changed to lowercase because
+                // DNS is case insensitive.
+                value = value.toLowerCase();
+            }
+
+            result.push(`${headerName}:${value}`);
+        }
+
+        result.push("");
+        result.push(req.body);
+
+        return result.join("\n");
+    }
+
+
+    /**
+     * Make sure the signature header attributes have exactly the items
+     * we require for this version of a signature.
+     *
+     * @param {Object} kvPairs Acts like a map of key/values.
+     * @return {Promise.<*>}
+     */
+    function checkSignatureAttributesAsync(kvPairs) {
+        var kvPairKeys, missing, requiredAttributes;
+
+        kvPairKeys = Object.keys(kvPairs);
+        requiredAttributes = [
+            "access-code",
+            "signature",
+            "signed-headers"
+        ];
+
+        missing = util.arrayRequireItems(kvPairKeys, requiredAttributes);
+
+        if (missing.length) {
+            return errorResponse.rejectedPromiseAsync(`Missing required attributes: ${missing.join(", ")}`, "IHyLdqig");
+        }
+
+        missing = util.arrayRequireItems(requiredAttributes, kvPairKeys);
+
+        if (missing.length) {
+            return errorResponse.rejectedPromiseAsync(`Prohibited extra attributes: ${missing.join(", ")}`, "iqpoGKJk");
+        }
+
+        return promise.resolve();
+    }
+
+
+    /**
+     * We only allow one method, algorithm and encoding.
+     *
+     * @param {validateSignatureMiddleware~signatureDescription} desc
+     * @return {Promise.<*>}
+     */
+    function checkSignatureDescriptionAsync(desc) {
```

Comment:
On line 228, you used 'signatureDesc' as the param. For the sake of consistency, one should be changed.